File: DevExpress-Demo/CS/ChartsDemo.Wpf/Modules/DataBinding/MvvmStyleBindingViewModel.cs
using System.Collections.Generic;
using System.Windows.Media;
using DevExpress.Mvvm;
using DevExpress.Xpf.Charts;

namespace ChartsDemo {
    public class DailyWeatherViewModel {
        const string vostokStationName = "Vostok Station";
        const string deathValleyName = "Death Valley, NV";

        readonly List<WeatherItem> weather;

        public List<WeatherItem> Weather { get { return this.weather; } }

        public DailyWeatherViewModel() {
            ExtremeWeatherData data = new ExtremeWeatherData();
            List<WeatherRecord> valleyData = data.LoadDeathValleyData();
            List<WeatherRecord> vostokData = data.LoadVostokStationData();

            Palette palette = new OfficePalette();
            this.weather = new List<WeatherItem>() {
                new WeatherItem(valleyData, false, palette[1], deathValleyName),
                new WeatherItem(valleyData, true, palette[1], deathValleyName),
                new WeatherItem(vostokData, false, palette[0], vostokStationName),
                new WeatherItem(vostokData, true, palette[0], vostokStationName),
            };
        }

        public virtual IChartAnimationService ChartAnimationService { get { return null; } }
        public void OnModuleLoaded() {
            if (ChartAnimationService != null)
                ChartAnimationService.Animate();
        }
    }

    public class WeatherItem : BindableBase {
        public int AverageLineThickness {
            get { return GetProperty(() => AverageLineThickness); }
            set { SetProperty(() => AverageLineThickness, value); }
        }
        public List<WeatherRecord> Data { get; private set; }
        public bool IsAverageWeather { get; private set; }
        public Color Color { get; private set; }
        public string Name { get; private set; }

        public WeatherItem(List<WeatherRecord> data, bool isAverageWeather, Color color, string name) {
            Data = data;
            IsAverageWeather = isAverageWeather;
            Color = color;
            Name = name;
            AverageLineThickness = 2;
        }
    }
}

File: DevExpress-Demo/VB/ChartsDemo.Wpf/Modules/DataBinding/MvvmStyleBindingViewModel.vb
Imports System.Collections.Generic
Imports System.Windows.Media
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.Charts

Namespace ChartsDemo

    Public Class DailyWeatherViewModel

        Const vostokStationName As String = "Vostok Station"

        Const deathValleyName As String = "Death Valley, NV"

        Private ReadOnly weatherField As List(Of WeatherItem)

        Public ReadOnly Property Weather As List(Of WeatherItem)
            Get
                Return weatherField
            End Get
        End Property

        Public Sub New()
            Dim data As ExtremeWeatherData = New ExtremeWeatherData()
            Dim valleyData As List(Of WeatherRecord) = data.LoadDeathValleyData()
            Dim vostokData As List(Of WeatherRecord) = data.LoadVostokStationData()
            Dim palette As Palette = New OfficePalette()
            weatherField = New List(Of WeatherItem)() From {New WeatherItem(valleyData, False, palette(1), deathValleyName), New WeatherItem(valleyData, True, palette(1), deathValleyName), New WeatherItem(vostokData, False, palette(0), vostokStationName), New WeatherItem(vostokData, True, palette(0), vostokStationName)}
        End Sub

        Public Overridable ReadOnly Property ChartAnimationService As IChartAnimationService
            Get
                Return Nothing
            End Get
        End Property

        Public Sub OnModuleLoaded()
            If ChartAnimationService IsNot Nothing Then ChartAnimationService.Animate()
        End Sub
    End Class

    Public Class WeatherItem
        Inherits BindableBase

        Private _Data As List(Of ChartsDemo.WeatherRecord), _IsAverageWeather As Boolean, _Color As Color, _Name As String

        Public Property AverageLineThickness As Integer
            Get
                Return GetProperty(Function() Me.AverageLineThickness)
            End Get

            Set(ByVal value As Integer)
                SetProperty(Function() AverageLineThickness, value)
            End Set
        End Property

        Public Property Data As List(Of WeatherRecord)
            Get
                Return _Data
            End Get

            Private Set(ByVal value As List(Of WeatherRecord))
                _Data = value
            End Set
        End Property

        Public Property IsAverageWeather As Boolean
            Get
                Return _IsAverageWeather
            End Get

            Private Set(ByVal value As Boolean)
                _IsAverageWeather = value
            End Set
        End Property

        Public Property Color As Color
            Get
                Return _Color
            End Get

            Private Set(ByVal value As Color)
                _Color = value
            End Set
        End Property

        Public Property Name As String
            Get
                Return _Name
            End Get

            Private Set(ByVal value As String)
                _Name = value
            End Set
        End Property

        Public Sub New(ByVal data As List(Of WeatherRecord), ByVal isAverageWeather As Boolean, ByVal color As Color, ByVal name As String)
            Me.Data = data
            Me.IsAverageWeather = isAverageWeather
            Me.Color = color
            Me.Name = name
            AverageLineThickness = 2
        End Sub
    End Class
End Namespace
